controllers/operands/kubevirt.go

```diff
@@ -421,6 +428,19 @@ func getKVConfig(hc *hcov1beta1.HyperConverged) (*kubevirtcorev1.KubeVirtConfigu
 
 	seccompConfig := getKVSeccompConfig()
 
+	if hc.Spec.FeatureGates.PrimaryUserDefinedNetworkBinding != nil && *hc.Spec.FeatureGates.PrimaryUserDefinedNetworkBinding {
+		if hc.Spec.NetworkBinding == nil {
+			hc.Spec.NetworkBinding = make(map[string]kubevirtcorev1.InterfaceBindingPlugin)
+		}
+
+		var sidecarImage string
+		var ok bool
+		if sidecarImage, ok = os.LookupEnv(hcoutil.PrimaryUDNImageEnvV); !ok {
+			return nil, errors.New("failed to get primary UDN image env Var")
+		}
+		hc.Spec.NetworkBinding[primaryUDNNetworkBindingName] = primaryUserDefinedNetworkBinding(sidecarImage)
```

Comment:
Why do we need to modify the HyperConverged CR? we'll need really good reason for that. the spec is for the user.

Reply:
We can't directly change the kubevirt CR. If we do, HCO will reconcile back to what's on hc.Spec.NetworkBinding..

Reply:
HCO currently just copies the `NetworkBinding` map to KV. I think we can change this code with the new logic.

e.g., instead of 
```go
		NetworkConfiguration: &kubevirtcorev1.NetworkConfiguration{
			NetworkInterface: string(kubevirtcorev1.MasqueradeInterface),
			Binding:          hc.Spec.NetworkBinding,
		},
```
do something like 
```go
		NetworkConfiguration: &kubevirtcorev1.NetworkConfiguration{
			NetworkInterface: string(kubevirtcorev1.MasqueradeInterface),
			Binding:          getNetworkBinding(hc),
		},
```

Reply:
and the new logic will get into the new function `getNetworkBinding`

Reply:
DONE